controllers/machinehealthcheck_controller.go

```diff
@@ -259,9 +262,57 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 		} else {
-			logger.Info("Target has failed health check, marking for remediation", "target", t.string(), "reason", condition.Reason, "message", condition.Message)
-			conditions.MarkFalse(t.Machine, clusterv1.MachineOwnerRemediatedCondition, clusterv1.WaitingForRemediationReason, clusterv1.ConditionSeverityWarning, "MachineHealthCheck failed")
+			if m.Spec.RemediationTemplate != nil {
+				// If external remediation request already exists,
+				// return early
+				if r.externalRemediationRequestExists(ctx, m, t.Machine.Name) {
+					return ctrl.Result{}, nil
+				}
+
+				cloneOwnerRef := &metav1.OwnerReference{
+					APIVersion: clusterv1.GroupVersion.String(),
+					Kind:       "Machine",
+					Name:       t.Machine.Name,
+					UID:        t.Machine.UID,
+				}
+
+				from, err := external.Get(ctx, r.Client, m.Spec.RemediationTemplate, t.Machine.Namespace)
+				if err != nil {
+					conditions.MarkFalse(m, clusterv1.ExternalRemediationTemplateAvailable, clusterv1.ExternalRemediationTemplateNotFound, clusterv1.ConditionSeverityError, "External Remediation Template not found")
+					return ctrl.Result{}, errors.Wrapf(err, "error retrieving remediation template %v %q for machine %q in namespace %q within cluster %q", m.Spec.RemediationTemplate.GroupVersionKind(), m.Spec.RemediationTemplate.Name, t.Machine.Name, t.Machine.Namespace, m.Spec.ClusterName)
+				}
+
+				generateTemplateInput := &external.GenerateTemplateInput{
+					Template:    from,
+					TemplateRef: m.Spec.RemediationTemplate,
+					Namespace:   t.Machine.Namespace,
+					ClusterName: t.Machine.ClusterName,
+					OwnerRef:    cloneOwnerRef,
+				}
+				to, err := external.GenerateTemplate(generateTemplateInput)
+				if err != nil {
+					return ctrl.Result{}, errors.Wrapf(err, "failed to create template for remediation request %v %q for machine %q in namespace %q within cluster %q", m.Spec.RemediationTemplate.GroupVersionKind(), m.Spec.RemediationTemplate.Name, t.Machine.Name, t.Machine.Namespace, m.Spec.ClusterName)
+				}
+
+				// Set the Remediation Request to match the Machine name, the name is used to
+				// guarantee uniqueness between runs. A Machine should only ever have a single
+				// remediation object of a specific GVK created.
+				//
+				// NOTE: This doesn't guarantee uniqueness across different MHC objects watching
+				// the same Machine, users are in charge of setting health checks and remediation properly.
+				to.SetName(t.Machine.Name)
+
+				logger.Info("Target has failed health check, creating an external remediation request", "remediation request name", to.GetName(), "target", t.string(), "reason", condition.Reason, "message", condition.Message)
+				// Create the external clone.
+				if err := r.Client.Create(ctx, to); err != nil {
+					return ctrl.Result{}, errors.Wrapf(err, "error creating remediation request for machine %q in namespace %q within cluster %q", t.Machine.Name, t.Machine.Namespace, t.Machine.ClusterName)
```

Comment:
Either here or in a follow-up PR: we need a condition that the user can see that creating the remediation request CR failed.

Reply:
This probably makes sense, considering that at this stage we could face a problem if the template is wrong or if the ExternalRemediation CRD is not installed.
This would be nice in this PR, but I won't consider this blocking. If this is not going in this PR, please open an issue to track this.

Reply:
@ncdc @fabriziopandini let's have it in this PR then. How about adding `ExternalRemediationAvailable` condition and set it to false here if creating the remediation request failed?

Reply:
+1 for me

Reply:
@fabriziopandini @ncdc I added the new condition for this.